pkg/gadgets/run/tracer/tracer.go

```diff
@@ -401,13 +411,80 @@ func (t *Tracer) processEventFunc(gadgetCtx gadgets.GadgetContext) func(data []b
 			}
 			timestampsOffsets = append(timestampsOffsets, member.Offset.Bytes())
 		}
+
+		btfSpec, err := btf.LoadKernelSpec()
+		if err != nil {
+			logger.Warnf("Kernel BTF information not available. Enums won't be resolved to strings")
+		}
+
+		if enum, ok := member.Type.(*btf.Enum); ok {
+			if btfSpec != nil {
+				kernelEnum := &btf.Enum{}
+				if err = btfSpec.TypeByName(enum.Name, &kernelEnum); err == nil {
+					// Use kernel enum if found
+					enum = kernelEnum
+				}
+			}
+
+			var getter func(data []byte) uint64
+			typ := simpleTypeFromBTF(member.Type)
+			switch typ.Kind {
```

Comment:
Can `simpleTypeFromBTF` return `nil`? In this case, reading `typ.Kind` would panic.

Reply:
I don't think it can return `nil`, it's called only for enums and all possibilities are handled in https://github.com/inspektor-gadget/inspektor-gadget/blob/2cf1a9e0cb47f8e627c6ab898e9b233922850bc0/pkg/gadgets/run/tracer/run.go#L746-L770

Reply:
I think `typedMember.Size` is unmarshalled from the ELF file (from the BTF section) and I am not sure if the cilium/ebpf package checks that the size of an enum is one of those values 1,2,4,8. I tried to find that check but I didn't find it.

https://github.com/cilium/ebpf/blob/main/btf/btf_types.go#L165


It should be fine for valid ebpf/ELF files, but we should ensure IG does not panic for invalid or malicious content.

Reply:
Makes sense. Added a check to verify this condition.